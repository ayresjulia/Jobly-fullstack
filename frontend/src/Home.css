.Home-body {
	font-size: 1.5rem;
	text-align: center;
	margin-top: 180px;
}
