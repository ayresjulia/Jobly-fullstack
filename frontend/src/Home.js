import React from "react";
import "./Home.css";

/** Home page. If user is logged in, welcomes them by username. */

const Home = ({ currentUser }) => {
	return (
		<div className="Home-body">
			<p className="font-weight-bold">Jobly</p>
			<p>All the jobs in one, convenient place.</p>
			{currentUser && (
				<p>
					Welcome back, <b>{currentUser.username}</b>!
				</p>
			)}
			{!currentUser && <p>Please, Log in or Sign up to begin.</p>}
		</div>
	);
};

export default Home;
