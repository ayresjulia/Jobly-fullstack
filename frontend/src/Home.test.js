import { render } from "@testing-library/react";
import Home from "./Home";
import { MemoryRouter } from "react-router";

it("renders without crashing", () => {
	render(
		<MemoryRouter>
			<Home />
		</MemoryRouter>
	);
});

it("matches snapshot", () => {
	const { asFragment } = render(
		<MemoryRouter>
			<Home />
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});

it("shows welcome text", () => {
	const { getByText } = render(
		<MemoryRouter>
			<Home />
		</MemoryRouter>
	);
	expect(getByText("Please, Log in or Sign up to begin.")).toBeInTheDocument();
});

it("shows welcome text", () => {
	let username = "testuser";
	const { getByText } = render(
		<MemoryRouter>
			<Home currentUser={username} />
		</MemoryRouter>
	);
	expect(getByText("Welcome back", { exact: false })).toBeInTheDocument();
});
