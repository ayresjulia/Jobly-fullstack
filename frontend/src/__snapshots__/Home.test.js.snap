// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Home-body"
  >
    <p
      class="font-weight-bold"
    >
      Jobly
    </p>
    <p>
      All the jobs in one, convenient place.
    </p>
    <p>
      Please, Log in or Sign up to begin.
    </p>
  </div>
</DocumentFragment>
`;
